### ISSUE TITLE
No CardDav error text on failed synchronisation

### ISSUE THREAD

**Issue by Nono-m0le** [the-djmaze/snappymail#344]
**Describe the bug**
I added my cardDav credentials on the settings menu. (didn't get any error but I guess that's normal as per the no 'test' button.
When I go back to the contacts book and try the synchronise from this server, I just got a popup with : 
Title : The domain name of the cardDav server (without subdomain)
Button: OK

**To Reproduce**
1. Go to Settings > Contact
2. add the url / user & password of your CardDav server 
3. select either "yes" or "Read only"
4. Go back to your contact book
5. click on the "burger menu", then Synchronization

Get the "error" windows which looks empty

**Expected behavior**
a clear error message

**Screenshots**
![image](https://user-images.githubusercontent.com/3726435/164808934-3856513e-1622-4bb7-8e12-501fd059f27a.png)

**Desktop (please complete the following information):**
 - Browser Firefox 99.0.1
 - SnappyMail Version 2.15.0

**Additional context**
the credentials or the password may be wrong on that case

**Comment by the-djmaze** [comment]
What does the browser debug window say in the network tab?

Like:
![afbeelding](https://user-images.githubusercontent.com/3752035/165060352-2bdfab20-50c5-4439-b008-8a6e83abb678.png)

**Comment by Nono-m0le** [comment]
weirdly, a 502 bad getway 🤷 
![image](https://user-images.githubusercontent.com/3726435/165373837-3903a847-b359-4759-8a5c-ba32651d0305.png)

**Comment by the-djmaze** [comment]
> weirdly, a 502 bad gateway

This is not weird.
When PHP-FPM is not responding fast enough, this could happen.
In that case, PHP error log should tell more about the issue.

**Comment by Nono-m0le** [comment]
Well, I have indeed php-fpm issues lately which I can't figure out why. (I did increase the timeout, execution & terminate time) without success. debug log doesn't help neither.
I know this is behond scope but, do you have any suggestion for nginx/php-fpm7.4 issues like this ?

**Comment by the-djmaze** [comment]
I'm using Apache with PHP-FPM and sometimes that also crashes.
Sometimes i just restart Apache or PHP-FPM (depending which one has issues).
It's only once in 3-4 months i have such issue.

**Comment by Nono-m0le** [comment]
I fixed my php-fpm issue, but still getting the same kind of error trying to sync the CardDav contacts ...
Is it possible to debug this further ?

**Comment by the-djmaze** [comment]
> Is it possible to debug this further ?

I might have an idea to make this a background task.

**Comment by the-djmaze** [comment]
Could you test with these:

[snappymail-2.16.1.tar.gz](https://github.com/the-djmaze/snappymail/files/8738769/snappymail-2.16.1.tar.gz)
[snappymail-2.16.1.zip](https://github.com/the-djmaze/snappymail/files/8738770/snappymail-2.16.1.zip)

**Comment by Nono-m0le** [comment]
somehow, it's better : 
![image](https://user-images.githubusercontent.com/3726435/169670323-f0f3fc82-e6f1-4cd1-a29c-0fff57ddcb49.png)

**Comment by the-djmaze** [comment]
Ah i forgot to change something after changing the code for it.
Could you try with the above commit of a fixed file?

**Comment by Nono-m0le** [comment]
got a "Contacts synchronization error" on the popup now.

**Comment by the-djmaze** [comment]
Well that's better. We have progress vs 502 Bad Gateway

**Comment by Nono-m0le** [comment]
Yup :)
I could basically close my ticket (and open another one) but it would be nice to have a small heads up or a debug log to got more information.
Is this error came from directly my nextcloud (which held my cardav server) or snappy itself ?

**Comment by the-djmaze** [comment]
Any luck with latest version?

**Comment by Nono-m0le** [comment]
I got the same error message, but without any further information.
